### PR TITLE
docs: ext-proc

### LINKED ISSUE #3456: docs: Ext Proc
*Description*:
>Describe the issue.

Add docs highlighting the Ext Proc feature

Relates to https://github.com/envoyproxy/gateway/issues/2546

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by guydc**
**What this PR does / why we need it**:
docs for the ext-proc feature
**Which issue(s) this PR fixes**:
Fixes #3456

### CODE REVIEW

File: site/content/en/latest/tasks/extensibility/ext-proc.md [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+```shell
+kubectl get httproute/myapp -o yaml
+```
+
+### Configuration
+
+Create a new EnvoyExtensionPolicy resource to configure the external processing service. This EnvoyExtensionPolicy targets the HTTPRoute
+"myApp" created in the previous step. It calls the GRPC external processing service "grpc-ext-proc" on port 9002 for
+processing. The `processingMode` struct defines that headers of requests and responses will be sent to the external processing server.
```
Comment (arkodg) [inline site/content/en/latest/tasks/extensibility/ext-proc.md:99]: may a line in here saying, `by default all request headers are sent, but in this example we are sending none` would be helpful

File: site/content/en/latest/tasks/extensibility/ext-proc.md [hunk 2]
```diff
@@ -96,7 +96,10 @@ kubectl get httproute/myapp -o yaml
 
 Create a new EnvoyExtensionPolicy resource to configure the external processing service. This EnvoyExtensionPolicy targets the HTTPRoute
 "myApp" created in the previous step. It calls the GRPC external processing service "grpc-ext-proc" on port 9002 for
-processing. The `processingMode` struct defines that headers of requests and responses will be sent to the external processing server.
+processing. 
+
+By default, requests and responses are not sent to the external processor. The `processingMode` struct is used to define what should be sent to the external processor.
```
Comment (arkodg) [inline site/content/en/latest/tasks/extensibility/ext-proc.md:101]: aren't request headers sent by default ?
https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_proc/v3/processing_mode.proto#envoy-v3-api-msg-extensions-filters-http-ext-proc-v3-processingmode

Comment (arkodg) [reply site/content/en/latest/tasks/extensibility/ext-proc.md:101]: maybe we should explicitly mention `body` so readers like me dont confuse it with `headers`

Comment (guydc) [reply site/content/en/latest/tasks/extensibility/ext-proc.md:101]: yes, but not in EG. We wanted to simplify the API (avoid explicitly mentioning header processing modes) while still supporting opt-out for request and/or response processing and explicit body processing mode configuration. 

To accomodate this, the API currently behaves like this:
1. if request/response is set to an empty struct - headers are sent. 
2. if request/response is configured with a body - headers and body are sent.
3. is request/response is not configured at all - opt-out of calling the processor. 

I can extend the example to also include body forwarding in one of the flows to help explain how this works.

Comment (arkodg) [reply site/content/en/latest/tasks/extensibility/ext-proc.md:101]: thanks for explaining this, in that case, can we update above text mentioning that request headers will be sent ?

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]

**Pull request merged**